src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,83 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * List to keep tasks.
+ */
+class TaskList {
+
+    private ArrayList<Task> listOfTasks = new ArrayList<>();
+
+    /**
+     * Adds a task to the list.
+     *
+     * @param task Task to be added.
+     */
+    public void addTask(Task task) {
+        listOfTasks.add(task);
+    }
+
+    /**
+     * Returns size of the list.
+     *
+     * @return Size of the list.
+     */
+    public int getSize() {
+        return listOfTasks.size();
+    }
+
+    /**
+     * Deletes a task at given position.
+     *
+     * @param i position of the task to be deleted.
+     * @throws DukeException when the task position is non-existent.
+     */
+    public void deleteTask(int i) throws DukeException {
+        if (listOfTasks.size() <= i) {
+            throw DukeException.IDTOOBIG;
+        }
+
+        listOfTasks.remove(i);
+    }
+
+    /**
+     * Get the task at given position
+     *
+     * @param i position of the task.
+     * @return the task at position.
+     * @throws DukeException when the task position is non-existent.
+     */
+    public Task getTask(int i) throws DukeException {
```

Comment:
Would it be possible to abstract this out? You could potentially have methods to mark/unmark in `TaskList` class so as to avoid having to pass `Task` objects around 😄